src/integrations/prefect-email/prefect_email/credentials.py

```diff
@@ -182,13 +190,17 @@ def example_get_server_flow():
         if smtp_type == SMTPType.INSECURE:
             server = SMTP(smtp_server, smtp_port)
         else:
-            context = ssl.create_default_context()
+            context = (
+                ssl._create_unverified_context(protocol=ssl.PROTOCOL_TLS_CLIENT)
+                if self.unverified_context
+                else ssl.create_default_context()
+            )
             if smtp_type == SMTPType.SSL:
                 server = SMTP_SSL(smtp_server, smtp_port, context=context)
             elif smtp_type == SMTPType.STARTTLS:
                 server = SMTP(smtp_server, smtp_port)
                 server.starttls(context=context)
-            if self.username is not None:
-                server.login(self.username, self.password.get_secret_value())
+        if self.username is not None:
+            server.login(self.username, self.password.get_secret_value())
```

Comment:
Can you add a warning when a username and password is used with `SMTPType.Insecure` or if `verify=False`? In general, users shouldn't send credentials in plain text or when certificates are not verified.